sdk/python/kfp/client/client.py

```diff
@@ -709,6 +710,8 @@ def run_pipeline(
                 is ``True`` for all tasks by default. If set, the
                 setting applies to all tasks in the pipeline (overrides the
                 compile time settings).
+            cache_key (optional): Customized cache key for this task.
+                If set, the cache_key will be used as the key for the task's cache.
```

Comment:
Could you elaborate on why we would need a cache key? This doc string isn't providing a clear enough idea.

I would suggest adding some description to the PR, and also linking the related GitHub issue.

Reply:
Thanks for reviewing!

This is for Issue https://github.com/kubeflow/pipelines/issues/11328
Doc: https://docs.google.com/document/d/1oNgYyFYondaVSFf9Pd3Q9uVzaqBr5wrOMHKgX9MMa78/edit?tab=t.0

(Also updated this information in description)